File: backend/app/schemas/argus.py
from app.schemas.common import CamelIn, CamelOut


class ChatMessageIn(CamelIn):
    role: str  # "user" | "assistant"
    content: str


class RepoRef(CamelIn):
    owner: str
    repo: str


class ChatIn(CamelIn):
    messages: list[ChatMessageIn]
    repo: RepoRef | None = None


class ChatOut(CamelOut):
    reply: str

File: src/lib/argus.ts
import { apiFetch } from "@/lib/api";

export interface ArgusMessage {
  role: "user" | "assistant";
  content: string;
}

export interface RepoRef {
  owner: string;
  repo: string;
}

/** Envia o histórico da conversa (e o repo conectado) ao agente. */
export function enviarArgus(messages: ArgusMessage[], repo?: RepoRef | null): Promise<{ reply: string }> {
  return apiFetch<{ reply: string }>("/api/argus/chat", {
    method: "POST",
    body: JSON.stringify({ messages, repo: repo ?? null }),
  });
}
